contracts/serverless-v2/Executors.sol

```diff
@@ -237,44 +237,75 @@ contract Executors is
     ) internal {
         executors[_executor].enclaveAddress = _enclaveAddress;
         executors[_executor].jobCapacity = _jobCapacity;
-        executors[_executor].status = true;
         
-        emit ExecutorRegistered(_executor, _enclaveAddress);
+        emit ExecutorRegistered(_executor, _jobCapacity);
     }
 
-    function _deregisterExecutor(
+    function _drainExecutor(
         address _executor
     ) internal {
-        if(!executors[_executor].status)
-            revert ExecutorsAlreadyInitiatedDeregister();
+        if(executors[_executor].draining)
+            revert ExecutorsAlreadyDrained();
 
-        executors[_executor].status = false;
+        executors[_executor].draining = true;
 
         // remove node from the tree
         _deleteIfPresent(_executor);
 
-        if(executors[_executor].activeJobs == 0)
-            _completeDeregister(_executor);
-        else
-            emit ExecutorDeregisterInitiated(_executor);
+        emit ExecutorsExecutorDrained(_executor);
+    }
+
+    function _reviveExecutor(
+        address _executor
+    ) internal {
+        Executor memory executorNode = executors[_executor];
+        if(!executorNode.draining)
+            revert ExecutorsAlreadyRevived();
+
+        executors[_executor].draining = false;
+
+        // insert node in the tree
+        if(executorNode.stakeAmount >= MIN_STAKE_AMOUNT && 
+            executorNode.activeJobs != executorNode.jobCapacity
```

Comment:
can you use `<` instead of `!=`

Reply:
sure, will update all the similar occurences.